accepted/2024/net10.md

```diff
@@ -0,0 +1,75 @@
+# Parsing `net10`
+
+**Owner** [Immo Landwerth](https://github.com/terrajobst)
+
+NuGet's framework name parser is fairly old and stems from a world where only
+.NET Framework exists. In fact, the early versions of NuGet didn't even support
+targeting different versions of .NET Framework.
+
+In this world, version numbers were represented without periods, for example,
+`net20` would be .NET Framework 2.0 while `net35` referred to .NET Framework
+3.5.
+
+And here lies the problem: next year we're shipping .NET 10. It's reasonable for
+people to put `net10` in the project file. This will produce some interesting
+error messages:
+
+> **error MSB3645**: .NET Framework v3.5 Service Pack 1 was not found. In order
+> to target ".NETFramework,Version=v1.0", .NET Framework v3.5 Service Pack 1 or
+> later must be installed.
+>
+> **error MSB3644**: The reference assemblies for .NETFramework,Version=v1.0
+> were not found. To resolve this, install the Developer Pack (SDK/Targeting
+> Pack) for this framework version or retarget your application. You can
+> download .NET Framework Developer Packs at
+> https://aka.ms/msbuild/developerpacks
+
+It's not exactly terrible, but many people will find this confusing as it's not
+obvious that the fix is to change the framework from `net10` to `net10.0`.
+
+This document proposes how to address this.
+
+## Stakeholders and Reviewers
+
+* NuGet team
+* MSBuild/SDK team
+* VS project system team
+
+## Design
+
+Considering that .NET Framework 1.0 is out of support for a while and Visual
+Studio doesn't even support building for anything older than .NET Framework 2.0,
+it seems very doable to make `net10` mean .NET 10. The question is what happens
+with all the other frameworks, such as `net20`, `net452` etc. Quite a few of
+those are still supported and actively used.
+
+Here is the proposal:
+
+* Make `net10` mean .NET 10 and `net11` mean .NET 11 but keep all higher
+  versions as-is (`net20` still means .NET Framework 2.0).
```

Comment:
this seems backwards to me. if the goal is to eventually guide people to always use periods then I don't think we should complicate our lives by allowing `net10` and potentially cause more issues in downstream tools that parse the version.

I'd change the rule to:
1. Warn about missing dot for (currently) non-ambigious versions like net20, net472
2. Error for ambigious versions like net10, net11.

And change to error everywhere in net12 like you suggested.

Reply:
> 1. Warn about missing dot for (currently) non-ambigious versions like net20, net472

We don't have any alternatives to `net20` or `net472` to fix the warning, do we? If not, we shouldn't warn for those.

Reply:
you can use `net2.0` and `net4.7.2` instead.

Reply:
TIL

Reply:
Agreed. I've changed the proposal.